components/style/context.rs

```diff
@@ -120,17 +121,22 @@ pub struct TraversalStatistics {
     pub elements_matched: u32,
     /// The number of cache hits from the StyleSharingCache.
     pub styles_shared: u32,
+    /// Time spent in the traversal, in milliseconds.
+    pub traversal_time_ms: f64,
 }
 
 /// Implementation of Add to aggregate statistics across different threads.
 impl<'a> Add for &'a TraversalStatistics {
     type Output = TraversalStatistics;
     fn add(self, other: Self) -> TraversalStatistics {
+        debug_assert!(self.traversal_time_ms == 0.0 && other.traversal_time_ms == 0.0,
+                      "traversal_time_ms should be set at the end by the caller");
```

Comment:
Seems to me this shouldn't live in `TraversalStatistics`, though I see why you're doing it.